servicebroker/lib/servicebroker.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var async = require('asyncawait/async');
+var await = require('asyncawait/await');
+var model = require(path.join(__dirname, './model'))();
+
+var ServiceBroker = function ServiceBroker() {};
+ServiceBroker.prototype.getCatalog = function() {
+  var catalog = JSON.parse(
+    fs.readFileSync(path.join(__dirname, '../config/catalog.json'), 'utf8'));
+  return catalog;
+};
+
+ServiceBroker.prototype.provisionService = async(function(serviceId, orgId, spaceId) {
```

Comment:
This is very confusing for many reasons.

1. We are making the whole thing async, but the actual findOrCreate which is async we made synchronous. If we plan to make this async, then don't bother wrapping.
2. We are returning status codes when we do require this to handle the http request. It is mixing ideas.

Let us follow the pattern that the API component has.

This file should go into a routes subdirectory and named serviceinstances.js
The HTTP handler and DB code should be co-mingled. There is no point in having the separation when we have no special business logic.